packages/cli/src/workflow-runner.ts

```diff
@@ -286,12 +295,54 @@ export class WorkflowRunner {
 					data.executionData,
 				);
 				workflowExecution = workflowExecute.processRunExecutionData(workflow);
+			} else if (data.preferredTrigger?.data && data.startNodes && !data.destinationNode) {
+				this.logger.debug(
+					`Execution ID ${executionId} had preferredTrigger. Starting from that trigger.`,
+					{ executionId },
+				);
+				const startNodes = data.startNodes.map((data) => {
+					const node = workflow.getNode(data.name);
+					a.ok(node, `Could not find a node named "${data.name}" in the workflow.`);
+					return node;
+				});
+
+				const { nodeExecutionStack, waitingExecution, waitingExecutionSource } =
+					recreateNodeExecutionStack(
+						filterDisabledNodes(DirectedGraph.fromWorkflow(workflow)),
+						new Set(startNodes),
+						{ [data.preferredTrigger.name]: [data.preferredTrigger.data] },
+						data.pinData ?? {},
+					);
+				const executionData: IRunExecutionData = {
+					resultData: {
+						runData: data.runData ?? {
+							[data.preferredTrigger.name]: [data.preferredTrigger.data],
+						},
+						pinData,
+					},
+					executionData: {
+						contextData: {},
+						metadata: {},
+						nodeExecutionStack,
+						waitingExecution,
+						waitingExecutionSource,
+					},
+				};
+
+				const workflowExecute = new WorkflowExecute(additionalData, 'manual', executionData);
+				workflowExecution = workflowExecute.processRunExecutionData(workflow);
 			} else if (
 				data.runData === undefined ||
 				data.startNodes === undefined ||
 				data.startNodes.length === 0
 			) {
 				// Full Execution
+				// TODO: When the old partial execution logic is removed this block can
+				// be removed and the previous one can be merged into
+				// `workflowExecute.runPartialWorkflow2`.
+				// Partial executions then require either a destination node from which
+				// everything else can be derived, or a preferredTrigger with
+				// triggerData. Full Execution
```

Comment:
This block is not necessary for the new partial execution flow. As soon as we default to the new flow and remove the old flow we can fold the code for handling preferred triggers into the partial execution function in `WorkflowExecute`